docgen/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/provider"
+	"github.com/olekukonko/tablewriter"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func main() {
+	docsPath := "docs/resources"
+
+	resources := provider.Provider().ResourcesMap
+
+	for name, resource := range resources {
+		shortName := strings.TrimPrefix(name, "snowflake_")
+
+		f, err := os.Create(path.Join(docsPath, fmt.Sprintf("%s.md", shortName)))
+		if err != nil {
+			panic(err)
+		}
+		defer f.Close()
+
+		f.WriteString(fmt.Sprintf("\n# %s\n\n", name))
+
+		f.WriteString("<!-- These docs are auto-generated by code in ./docgen, run by with make docs. Manual edits will be overwritten. -->\n")
```

Comment:
**[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Error return value of `f.WriteString` is not checked (errcheck)